CDCgov/prime-reportstream
# 1213 Add CSV to HL7 integration tests

This PR adds an integration test for the conversion from CSV to HL7 to verify that the generation of HL7 data is not broken with changes to the baseline.  This works in the same manner as the HL7 to CSV tests (formerly known as the ObservationMessageTest), where you have a folder with CSV input data and a companion HL7 file with the expected HL7 output data.  The tests will generate errors and warnings as follows:
Errors are generated when:
1. The number of reports is different
2. A segment in the expected values does not exist in the actual values
3. A expected component value does not match the actual value

Warnings are generated when:
1. A component actual value exists, but no expected value.

Test steps:
1. Build the baseline with `./gradlew package`
2. Verify the test results for both the test and testIntegration test suite.  See build/reports/test.
3. You can run the integration test only with `./gradlew testIntegration` and force it with `./gradlew testIntegration -Pforcetest`

## Changes
- Create a new test suite called integration tests that runs separate from the unit tests.  We will be adding tests like quicktest to this suite
- Publish the test reports in Github Actions and the PR
- Created CSV to HL7 test 
- Renamed the old ObservationMessageTest to HL7 to CSV conversion tests.
- Increased the wait time for the DEV settings container, otherwise I could not run the router correctly
- Added extra check to DateTimeOffsetMapper as it was throwing an exception when a date field is empty

## Checklist

### Testing

### Security

### Process

## Security
*List potential security threats and mitigations in the PR*

## Fixes
- #1213

## To Be Done
